sampleapp/src/main/scala/de/up/hpi/informationsystems/sampleapp/dactors/Customer.scala

```diff
@@ -97,34 +97,33 @@ object Customer {
 
       case AddStoreVisit.Request(storeId: Int, time: ZonedDateTime, amount: Double, fixedDiscount: Double, varDiscount: Double) =>
         addStoreVisit(storeId, time, amount, fixedDiscount, varDiscount) match {
-          case Success(_) => sender() ! AddStoreVisit.Success(Seq.empty)
+          case Success(_) => sender() ! AddStoreVisit.Success(Relation(Seq.empty))
           case Failure(e) => sender() ! AddStoreVisit.Failure(e)
         }
 
       case Authenticate.Request(passwordHash) =>
         if (authenticate(passwordHash)) {
-          sender() ! Authenticate.Success(Seq.empty)
+          sender() ! Authenticate.Success(Relation(Seq.empty))
```

Comment:
We may have an alias for this:
```scala
val rel = Relation(Seq.empty)
val rel2 = Relation.empty
assert rel1 == rel2 // true

// impl:
object Relation {
  [...]
  val empty: Relation = new TransientRelation(Try(Seq.empty))
}
```

Reply:
We don't yet though, right? I'm adding it as a `def` instead of `val` - hope that's alright with you.

Reply:
Yes, but i intentionally used `val` as we only need to create this object once and can reuse it. You can create a `private lazy val int_empty: Relation` if you like and then reference it from your `def empty`.